main.cpp

```diff
@@ -126,6 +128,12 @@ void parseArgs(int argc, char **argv, Settings &settings) {
             case 'i':
                 settings.info = true;
                 break;
+            case 'f':
+                {
+                    ifstream ifs(optarg);
+                    parseCfgFile(ifs, settings);
```

Comment:
Settings from the default config files will be present. Any settings from the defaults that are not present in this file will still remain in the settings. Also, depending on invocation order, command line arguments may be overridden by the settings file.

Possible solutions:
* destroy existing settings before reading -f
* do not read default files when -f
Both would require a "two pass" approach to reading arguments, the first lookng for -f and reading it prior to parsing the rest of the command line arguments.